statsd/statsd_test.go

```diff
@@ -165,6 +165,23 @@ func TestCloseRace(t *testing.T) {
 	wg.Wait()
 }
 
+func TestCloseWithClientAlreadyClosed(t *testing.T) {
+	c, err := New("localhost:8125")
+	c.isClosed = true
+	assert.NoError(t, err)
+	c.Close()
+	assert.True(t, c.IsClosed())
+	c.sender.close()
+}
```

Comment:
I think testing a double `Close` would be better:
```suggestion
func TestCloseWithClientAlreadyClosed(t *testing.T) {
	c, err := New("localhost:8125")
	assert.NoError(t, err)
	assert.False(t, c.IsClosed())
	
	assert.NoError(t, c.Close())
	assert.True(t, c.IsClosed())
	
	assert.NoError(t, c.Close())
        assert.True(t, c.IsClosed())
}
```